### DISCUSSION TITLE
Making separated form schema.

### DISCUSSION THREAD

**Discussion by Nuhel** [filamentphp/filament#4689] [Ideas]
Filament support to attach create form with select input, and this one of the great feature I must say.

But the issue i face is, i need to duplicate the form schema, where i use a select field with create form, it its sometime breaking, cause if i add modify validation rules, or modify a input field, i need to change this every where.

So what I am proposing is, what if there is a schema or form schema folder and it contains the form schema? And we can use it everywhere using static method?

And yes we can use a new option in command like --separate-schema to pass to make this happen, otherwise it will generate the resource, how it's generating now.

**Comment by mikrosmile** [comment]
create public function getFormSchema():array 
{
return [];
}

in desired CreatePage or Resource 

And inside form->schema([CreatePage::getFormSchema()])

**Reply by Nuhel** [reply]
> create public function getFormSchema():array { return []; }
> 
> in desired CreatePage or Resource
> 
> And inside form->schema([CreatePage::getFormSchema()])

Yes, but some time form get large , and its get hard to maintain in one page :).
That's why i offer this.

**Reply by mikrosmile** [reply]
Well, I created different functions with sections and pass across app sections I need 

For example, I have a user resource with many fields like name, surname, sex, birth etc etc but split it to sections like:
getFullNameSection 
getAddressSection 

And inside createOptionForm I pass only the fields from fullName, so I quickly can create user with just three fields ...

**Reply by Nuhel** [reply]
> Well, I created different functions with sections and pass across app sections I need
> 
> For example, I have a user resource with many fields like name, surname, sex, birth etc etc but split it to sections like: getFullNameSection getAddressSection
> 
> And inside createOptionForm I pass only the fields from fullName, so I quickly can create user with just three fields ...

Yes i do agree with you, What i said is if there is a large form, than the resource class get messy, That's why I like to separate the form schema from the resource class.

**Reply by mikrosmile** [reply]
Well, maybe a trait is what you're looking for ) 
Create a static form schemas, retrieve it whenever you need ... all in one place

**Comment by Z3d0X** [comment]
Theres nothing preventing you from doing this in your own application.

Resource class is already the class which contains the static methods used by Pages(LW components)
